crates/moving-average/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use std::collections::VecDeque;
```

Comment:
This is for the moving average by time; the two crates are based on fixed window size.  

This will be used across all indexer-grpc components(cache worker/data service/file store).

Reply:
Multiple places. We might be able to use an existing one sure.

Reply:
There are some differences (we're doing this based on time) so let's mark this as an optimization opportunity.